src/main/java/com/github/housepower/jdbc/data/type/complex/DataTypeDateTime64.java

```diff
@@ -0,0 +1,148 @@
+
+            Validate.isTrue(lexer.character() == ')');
+            return new DataTypeDateTime64("DateTime64(" + precision + ")", serverInfo);
+        }
+        return new DataTypeDateTime64("DateTime64", serverInfo);
+    }
+
+    @Override
+    public String name() {
+        return name;
+    }
+
+    @Override
+    public int sqlTypeId() {
+        return Types.TIMESTAMP;
+    }
+
+    @Override
+    public Object defaultValue() {
+        return DEFAULT_VALUE;
+    }
+
+    @Override
+    public Class javaTypeClass() {
+        return Timestamp.class;
+    }
+
+    @Override
+    public boolean nullable() {
+        return false;
+    }
+
+    @Override
+    public Object deserializeTextQuoted(SQLLexer lexer) throws SQLException {
+        Validate.isTrue(lexer.character() == 't');
+        Validate.isTrue(lexer.character() == 'o');
+        Validate.isTrue(lexer.character() == 'D');
+        Validate.isTrue(lexer.character() == 'a');
+        Validate.isTrue(lexer.character() == 't');
+        Validate.isTrue(lexer.character() == 'e');
+        Validate.isTrue(lexer.character() == 'T');
+        Validate.isTrue(lexer.character() == 'i');
+        Validate.isTrue(lexer.character() == 'm');
+        Validate.isTrue(lexer.character() == 'e');
+        Validate.isTrue(lexer.character() == '6');
+        Validate.isTrue(lexer.character() == '4');
+        Validate.isTrue(lexer.character() == '(');
+        Validate.isTrue(lexer.character() == '\'');
+        int year = lexer.numberLiteral().intValue();
+        Validate.isTrue(lexer.character() == '-');
+        int month = lexer.numberLiteral().intValue();
+        Validate.isTrue(lexer.character() == '-');
+        int day = lexer.numberLiteral().intValue();
+        Validate.isTrue(lexer.isWhitespace());
+        int hours = lexer.numberLiteral().intValue();
+        Validate.isTrue(lexer.character() == ':');
+        int minutes = lexer.numberLiteral().intValue();
+        Validate.isTrue(lexer.character() == ':');
+        int seconds = lexer.longLiteral().intValue();
+        Validate.isTrue(lexer.character() == '.');
+        int nanos = lexer.longLiteral().intValue();
+        Validate.isTrue(lexer.character() == '\'');
+        Validate.isTrue(lexer.character() == ')');
+
+        String timeStr = String.format("%04d-%02d-%02d %02d:%02d:%02d.%09d", year, month, day, hours, minutes, seconds, nanos);
+        LocalDateTime dateTime = LocalDateTime.parse(timeStr, dateTimeFormat);
+        return Timestamp.valueOf(dateTime);
+    }
+
+    @Override
+    public Object deserializeBinary(BinaryDeserializer deserializer) throws IOException {
+        long epochSeconds = deserializer.readInt();
+        long nanos = deserializer.readInt();
+        return Timestamp.valueOf(Instant.ofEpochSecond(epochSeconds, nanos).atZone(timeZone.toZoneId()).toLocalDateTime());
+    }
+
+    @Override
+    public void serializeBinary(Object data, BinarySerializer serializer) throws IOException {
```

Comment:
Could we  create a test case  to cover this method?

```
stmt.setObject(1, long_value)
stmt.execute...
```

Reply:
I added tests for min value of DateTime and max value of DateTime. Is it ok?
I am not understand about test case above, but there are some test cases in InsertComplexTypeITest which cover this method.

Reply:
And i rewrite binary serialization to make it more optimal